shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/variantcallers/HaplotypeCallerGvcf.scala

```diff
@@ -34,14 +37,82 @@ class HaplotypeCallerGvcf(val parent: Configurable) extends Variantcaller {
 
   def getGvcfs = gVcfFiles
 
+  val genderAwareCalling: Boolean = config("gender_aware_calling", default = false)
+  val haploidRegions: Option[File] = config("hap̦loid_regions")
+  val haploidRegionsMale: Option[File] = config("haploid_regions_male")
+  val haploidRegionsFemale: Option[File] = config("haploid_regions_female")
+
+  lazy val refernceSize = referenceDict.getReferenceLength
+
+  lazy val fractionMale: Double = calculateFraction(haploidRegions, haploidRegionsMale)
+  lazy val fractionFemale: Double = calculateFraction(haploidRegions, haploidRegionsFemale)
+  lazy val fractionUnknown: Double = calculateFraction(haploidRegions, None)
+
+  def calculateFraction(file1: Option[File], file2: Option[File]): Double = {
+    (file1.map(BedRecordList.fromFile(_)), file2.map(BedRecordList.fromFile(_))) match {
+      case (Some(l), None) => l.length.toDouble / refernceSize
+      case (None, Some(l)) => l.length.toDouble / refernceSize
+      case (Some(l1), Some(l2)) =>
+        BedRecordList.fromList(l1.allRecords ++ l2.allRecords).combineOverlap.length.toDouble / refernceSize
+      case (None, None) => 0.0
+    }
+  }
+
   override def fixedValues = Map("haplotypecaller" -> Map("emitRefConfidence" -> "GVCF"))
 
+  override def defaults = Map("haplotypecaller" -> Map(
+    "variant_index_type" -> "LINEAR",
+    "variant_index_parameter" -> 128000)
+  )
+
   def biopetScript() {
     gVcfFiles = for ((sample, inputBam) <- inputBams) yield {
-      val hc = gatk.HaplotypeCaller(this, List(inputBam), new File(outputDir, sample + ".gvcf.vcf.gz"))
-      hc.BQSR = inputBqsrFiles.get(sample)
-      add(hc)
-      sample -> hc.out
+      if (genderAwareCalling) {
+        val finalFile = new File(outputDir, sample + ".gvcf.vcf.gz")
+        val gender = genders.getOrElse(sample, Gender.Unknown)
+        val haploidBedFiles: List[File] = gender match {
+          case Gender.Female => haploidRegions.toList ::: haploidRegionsFemale.toList ::: Nil
+          case Gender.Male   => haploidRegions.toList ::: haploidRegionsMale.toList ::: Nil
+          case _             => haploidRegions.toList
+        }
+
+        val fraction: Double = gender match {
+          case Gender.Female => fractionFemale
+          case Gender.Male   => fractionMale
+          case _             => fractionUnknown
+        }
+
+        val haploidGvcf = if (haploidBedFiles.nonEmpty) {
+          val hc = gatk.HaplotypeCaller(this, List(inputBam), new File(outputDir, sample + ".haploid.gvcf.vcf.gz"))
+          hc.BQSR = inputBqsrFiles.get(sample)
+          hc.intervals = haploidBedFiles
+          hc.scatterCount = (hc.scatterCount * fraction).toInt
+          hc.sample_ploidy = Some(1)
+          add(hc)
+          Some(hc.out)
+        } else None
```

Comment:
It's probably better to give warning/error that gender-aware calling does not make sense/is impossible when there are no haploid bed files, in stead of just silently skipping the step.